pkg/controller/deployment/deployment_controller.go

```diff
@@ -576,6 +579,73 @@ func (dc *DeploymentController) syncRollingUpdateDeployment(deployment extension
 	return dc.syncDeploymentStatus(allRSs, newRS, deployment)
 }
 
+// proportionalScaling should run only on scaling events and not during the normal rollout process.
+func (dc *DeploymentController) proportionalScaling(deployment extensions.Deployment, shouldCreate bool) error {
+	newRS, oldRSs, err := dc.getAllReplicaSets(deployment, shouldCreate)
+	if err != nil {
+		return err
+	}
+	allRSs := controller.FilterActiveReplicaSets(oldRSs)
+	maybeMaxSurge := 0
+	if deployment.Spec.Strategy.Type == extensions.RollingUpdateDeploymentStrategyType {
+		maybeMaxSurge, err = intstrutil.GetValueFromIntOrPercent(&deployment.Spec.Strategy.RollingUpdate.MaxSurge, deployment.Spec.Replicas)
+		if err != nil {
+			return err
+		}
+	}
+
+	if newRS != nil {
+		// No old replica sets with pods; scale new replica set to the full count.
+		if len(controller.FilterActiveReplicaSets(oldRSs)) == 0 {
+			if _, _, err := dc.scaleReplicaSetAndRecordEvent(newRS, deployment.Spec.Replicas, deployment); err != nil {
+				return err
+			}
+			deployment.Status.EnactedReplicas = deployment.Spec.Replicas + maybeMaxSurge
```

Comment:
I thought EnactedReplicas was supposed to reflect the whole size of the deployment (.spec.replicas) plus any surge, so that we can identify scaling events on the deployment. Isn't that right?